### DISCUSSION TITLE
Should `Msg` implement `io.WriterTo`?

### DISCUSSION THREAD

**Discussion by inliquid** [wneessen/go-mail#3] [General]
Hi!

Many thanks for this library, specifically for `WriteToSendmail*` methods, this is exactly what I was looking for, and I didn't find it anywhere else. I think it makes sense to mention it in README.md. 

I have one small question regarding `Msg.Write` method. In original `gomail` lib and its forks it was named `WriteTo` so that it makes email object implement `io.WriterTo` interface. I'm not sure whether it gives anything, because you still can't use it in `io.Copy` until it implements `io.Reader`. Just curious if there was any specific reason to not implement `io.WriterTo`?

**Comment by wneessen** [comment]
> Many thanks for this library, specifically for `WriteToSendmail*` methods, this is exactly what I was looking for, and I didn't find it anywhere else. I think it makes sense to mention it in README.md.
> 
Glad to hear that people find it valuable. I'll add it to the readme.

> I have one small question regarding `Msg.Write` method. In original `gomail` lib and its forks it was named `WriteTo` so that it makes email object implement `io.WriterTo` interface. I'm not sure whether it gives anything, because you still can't use it in `io.Copy` until it implements `io.Reader`. Just curious if there was any specific reason to not implement `io.WriterTo`?

That's actually a good and valid point. I never thought about that tbh. I renamed it to just `Write()` for ease of readablity but I never considered that people might use it as part of the WriteTo interface. I'll believe it makes sense to rename it to `WriteTo()` and add an alias method `Write()` that returns to `WriteTo()` to not break implementations.

**Comment by wneessen** [comment]
@inliquid You mentioned `io.Copy` before. In what context would you want to use it? I was thinking about maybe adding a `msg.Read` method that satifies the `io.Reader` interface. Maybe write the whole message to a bufio and then return the byteslice so it could be used with `io.Copy`. Just curious if there's a specific need this could be helpful for

**Reply by inliquid** [reply]
I thought about using `io.Copy` for convenience reasons, such as dumping a message to log writer, but tbh this is not something I really need. I'm perfectly fine with current API =) However, as `Msg` now satisfies `io.WriterTo` it seems strange that you can't use  it as an argument for widely used `io.Copy`. I think adding this makes sense.

**Reply by wneessen** [reply]
io.Reader is now implemented via https://github.com/wneessen/go-mail/issues/11
